ci/Dockerfile

```diff
@@ -26,14 +26,21 @@ ENV PATH /root/.cargo/bin/:$PATH
 # add required rustup components
 RUN rustup component add rustfmt llvm-tools-preview clippy
 
+ENV CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
+
 # install build tools
-RUN cargo install cargo-llvm-cov cargo-nextest cargo-udeps cargo-hakari cargo-sort cargo-make cargo-cache sccache \
+RUN cargo install cargo-quickinstall --locked
+RUN cargo quickinstall cargo-binstall
+RUN cargo binstall -y --no-symlinks cargo-llvm-cov cargo-nextest cargo-udeps cargo-hakari cargo-sort cargo-make cargo-cache \
+    && cargo install sccache --locked \
     && cargo install --git https://github.com/risinglightdb/sqllogictest-rs --rev dbadddb --bin sqllogictest --locked \
     && cargo cache -a \
     && rm -rf "/root/.cargo/registry/index" \
     && rm -rf "/root/.cargo/registry/cache" \
     && rm -rf "/root/.cargo/git/db"
+RUN cargo uninstall cargo-quickinstall cargo-binstall
 
 ENV RUSTC_WRAPPER=sccache
 ENV SCCACHE_BUCKET=ci-sccache-bucket
-ENV CARGO_INCREMENTAL=0
\ No newline at end of file
+ENV CARGO_INCREMENTAL=0
```

Comment:
Hmm, so incremental is faster for from-scratch build after all? Why is that?

Reply:
What do you mean? 👀 This line is not changed and is disabling incremental